### ISSUE TITLE
Duplicate identifier "module-version" error with yanglint trying to parse ietf-yang-push-revision@2023-10-17.yang

### ISSUE THREAD

**Issue by jeremie6wind** [CESNET/libyang#2208]
Hello,
I got an error trying to parse new yang files.
Procedure to reproduce:

1) I created a directory containing the yang files related to subscriptions, from Netopeer2/modules:
ietf-datastores@2018-02-14.yang   ietf-network-instance@2019-01-21.yang          ietf-yang-push@2019-09-09.yang
ietf-interfaces@2018-02-20.yang   ietf-restconf@2017-01-26.yang        ietf-ip@2018-02-22.yang           ietf-subscribed-notifications@2019-09-09.yang  ietf-netconf-acm@2018-02-14.yang  ietf-yang-patch@2017-02-22.yang.

2)  I added these yang drafts:
ietf-yang-push-revision@2023-10-17.yang ietf-yang-revisions@2022-11-29.yang ietf-yang-semver@2022-09-13.yang
[ietf-yang-semver@2022-09-13.yang.txt](https://github.com/CESNET/libyang/files/14743978/ietf-yang-semver%402022-09-13.yang.txt)
[ietf-yang-revisions@2022-11-29.yang.txt](https://github.com/CESNET/libyang/files/14743979/ietf-yang-revisions%402022-11-29.yang.txt)
[ietf-yang-push-revision@2023-10-17.yang.txt](https://github.com/CESNET/libyang/files/14743980/ietf-yang-push-revision%402023-10-17.yang.txt)

3) Perform a yanglint(build from SHA: f4ba2ffa03e508a1e5378208fe576f9b360c33f8):
./yanglint ~/sshfs_root/yang_env/ietf-yang-push-revision@2023-10-17.yang

I get this error :
**libyang err : Duplicate identifier "module-version" of data definition/RPC/action/notification statement. (/ietf-yang-push-revision:{augment='/sn:subscriptions/sn:subscription/sn:target/yp:datastore/yp:selection-filter/yp:within-subscription'}/{uses='ypr:yang-push-module-version-config-list'}/ietf-yang-push-revision:module-version)**

Basically it seems it is not possible to augment both:

  augment "/sn:establish-subscription/sn:input/sn:target/sn:stream/sn:stream-filter/sn:within-subscription" {

  augment "/sn:establish-subscription/sn:input/sn:target/yp:datastore/yp:selection-filter/yp:within-subscription" {

with the same grouping.

Same error with both:

  augment "/sn:subscription-started/sn:target/sn:stream/sn:stream-filter/sn:within-subscription" {
   
  augment "/sn:subscription-started/sn:target/yp:datastore/yp:selection-filter/yp:within-subscription" {

BR,
J

**Comment by michalvasko** [comment]
I have improved the error message but I believe it is valid. Based on the new message and the RFC spec ([second bullet](https://datatracker.ietf.org/doc/html/rfc7950#page-48) at the top of the page). In practice, when parsing data, it would be impossible to distinguish between the two.
